### ISSUE TITLE
add inline ignore comment

### ISSUE THREAD

**Issue by Cielquan** [asottile/pyupgrade#622]
I have a py3.10 code base and run pyupgrade via pre-commit.

In this code base I use pydantic models which use
> standard library typing types as defined in PEP 484 to define complex objects.

[link to docs](https://pydantic-docs.helpmanual.io/usage/types/#typing-iterables)

So I have to define a model like this (example from docs):
```python
from typing import List

from pydantic import BaseModel


class Model(BaseModel):
    simple_list: list = None
    list_of_ints: List[int] = None
```

Which in turn gets modified by pyupgrade:
```diff
class Model(BaseModel):
    simple_list: list = None
-    list_of_ints: List[int] = None
+    list_of_ints: list[int] = None
```

And this results in a `RuntimeError` from pydantic.

To prevent this I currently use the workaround of skiping the model files whole which I luckily have in their own subpackage by ignoring them via the pre-commit config. But skiping them whole is a strong solution and I would rather skip single lines with something like a `# noqa` comment of some sort.

**Comment by MarcoGorelli** [comment]
you can use [keep-runtime-typing](https://github.com/asottile/pyupgrade#pep-585-typing-rewrites)

**Comment by Cielquan** [comment]
The skip is only needed for the model classes. Other classes or functions would also be skipped with this flag then, right? I would have to give up on annotation upgrades for everything then, which I would like to avoid.

I unfortunately forgot while writing the OP that I also (will) have pydantic models outside of the models subpackage, which only holds DB models. The codebase is a fastAPI app. But the issue only occurred with the DB models till now. So even having 2 entries in pre-commit config (One for the model subpackage with the flag and one for the rest without it.) won't cut it unfortunately.

**Comment by Cielquan** [comment]
As a temporary better solution than full ignore I have this now:
```yaml
  - repo: https://github.com/asottile/pyupgrade
    rev: e695ecd365119ab4e5463f6e49bea5f4b7ca786b # frozen: v2.31.0
    hooks:
      - id: pyupgrade
        name: pyupgrade (not models)
        args: ["--py310-plus"]
        exclude: .*\/models\/.*
      - id: pyupgrade
        name: pyupgrade (models)
        args: ["--py310-plus", "--keep-runtime-typing"]
        files: .*\/models\/.*
```

Which in turn results in:
```
pre-commit run pyupgrade --all-files
pyupgrade (not models)...................................................Passed
pyupgrade (models).......................................................Failed
- hook id: pyupgrade
- exit code: 1
- files were modified by this hook

Rewriting packages/swlp-api/app/models/user/user.py
Rewriting packages/swlp-api/app/models/user/user_role.py
```

So the `--keep-runtime-typing` flag does not apply correctly? Or is my config not right?

**Comment by asottile** [comment]
show your error? I suspect you're using python<3.9 but you've told pyupgrade you only support python>=3.10

**Comment by asottile** [comment]
but anyway -- this feature request will not be implemented -- there's already the `--keep-runtime-typing` option for broken things like pydantic

**Comment by Cielquan** [comment]
> show your error? I suspect you're using python<3.9 but you've told pyupgrade you only support python>=3.10

The package itself is set to python>=3.10 in its metadata.
But I did neither set `language_version` nor `default_language_version` in the pre-commit config. Will test this tomorrow.

> but anyway -- this feature request will not be implemented -- there's already the `--keep-runtime-typing` option for broken things like pydantic

What a bummer, would have been nice to skip at individual loc.

**Comment by asottile** [comment]
in the future please search the issue tracker -- #574

**Comment by Cielquan** [comment]
https://github.com/asottile/pyupgrade/issues/623#issuecomment-1090339004
> please continue the discussion in the original thread -- I don't think that matters, list[T] works fine for me in python3.10 with pydantic -- show your actual error

Well the issue was, that I created database models with sqlmodel (bridge between pydantic and sqlalchemy) and then alembic failed the automated database migration. After some further manual testing I found that the issue were some annotations `list[T]` and when I changed them to `typing.List[T]` the issue was gone.

Parallel to this github issue and #623 I further investigated the whole time and found out that it had something to do with `T` being `"T"`. `T` must be a string because of circular imports and thus using a `if tpying.TYPECHECKING:` guard. I do not use `from __future__ import annotations` because there were other issues with it.

In the end I just found the underlying issue, which is already fixed and only needs to be merged: https://github.com/samuelcolvin/pydantic/pull/3681

**Comment by asottile** [comment]
ok so as I suspected, not a pyupgrade bug but a bug in a third party tool -- if you would have led with the actual stacktrace you saw this would have been very obvious and would have saved us all time

in the future, please describe your actual problem rather than jumping to prescribing a solution -- I've put together a quick tutorial which outlines a strategy for making bug reports: https://youtu.be/ritp4gAqNMI

**Comment by antonagestam** [comment]
I thought I'd share here for anyone looking for workarounds (not a feature request 😉). I stumbled into the need of an ignore comment today because [there's a bug in mypy](https://github.com/python/mypy/issues/11098) where it won't work properly with builtin tuple. The workaround is to use `typing.Tuple`, but `pyupgrade --py310-plus --keep-runtime-typing` replaces that with `tuple`.

A workaround that both works with mypy and prevents pyupgrade from replacing it is to use string syntax with `typing.TypeAlias`. In my case I replaced this implicit type alias:

```python
_Shape2D = _HasShape2D | tuple[int, int] | Fraction
```

With this explicit one:

```python
_Shape2D: TypeAlias = "_HasShape2D | Tuple[int, int] | Fraction"
```
